src/components/MiddleEllipsis/MiddleEllipsis.tsx

```diff
@@ -1,18 +1,21 @@
 import { Box, BoxProps } from '@chakra-ui/react'
 import React from 'react'
-import ReactMiddleEllipsis from 'react-middle-ellipsis'
 
 type MiddleEllipsisProps = {
-  children: React.ReactNode
-} & ({ maxWidth: string } | { width: string }) &
-  BoxProps
+  address: string
+} & BoxProps
 
-export const MiddleEllipsis = ({ children, maxWidth, width, ...rest }: MiddleEllipsisProps) => {
+export const MiddleEllipsis = ({ address, ...rest }: MiddleEllipsisProps) => {
   return (
-    <Box whiteSpace='nowrap' maxWidth={maxWidth} width={width} {...rest}>
-      <ReactMiddleEllipsis>
-        <span>{children}</span>
-      </ReactMiddleEllipsis>
+    <Box whiteSpace='nowrap' {...rest}>
+      <span>{shortenAddress(address)}</span>
     </Box>
   )
 }
+
+export function shortenAddress(address: string, chars = 4): string {
+  if (address.length <= 40) {
```

Comment:
Magic number? @0xdef1cafe what do you think?
At least this fixes the regression on the from/to txs in trade history, just wondering about that magic 40

Reply:
lol, I chose 40 as google said that's how long the eth addresses are

Reply:
yeah - not a fan.

this should probably take a single object as an arg, like 
```
{
   value: string
   chars?: number
   isCryptoAddress?: boolean
}
```

and use the `isCryptoAddress` logic to decide how/when to shorten

Reply:
fixed it